packages/x-components/src/plugins/x-plugin.mixin.ts

```diff
@@ -37,20 +50,22 @@ export const createXComponentAPIMixin = (
  * Creates an object containing the API related to the {@link XBus}.
  *
  * @param bus - The global {@link XBus}.
- * @param xComponent - The root {@link XComponent} that the component that owns this API has.
+ * @param component - The component instance.
+ *
  * @returns An object containing the {@link XComponentBusAPI}.
  * @internal
  */
-export function getBusAPI(bus: XBus, xComponent: XComponent | undefined): XComponentBusAPI {
+export function getBusAPI(bus: XBus, component: PrivateExtendedVueComponent): XComponentBusAPI {
   return {
     emit: <Event extends XEvent>(
       event: Event,
       payload?: XEventPayload<Event>,
-      metadata: Omit<WireMetadata, 'moduleName'> = {}
+      metadata: Omit<WireMetadata, 'moduleName' | 'origin'> = {}
     ) => {
-      const moduleName = xComponent ? getXComponentXModuleName(xComponent) : null;
-      bus.emit(event, payload as any, { ...metadata, moduleName });
-      xComponent?.$emit(event, payload);
+      const rootComponent = component.xComponent;
```

Comment:
`rootComponent` is confusing, keep `xComponent` but I would do it directly in the next line:

`const moduleName = getXComponentXModuleName(component.xComponent) ?? null;`

Reply:
```
export function getXComponentXModuleName(component: Vue): XModuleName | null {
  return (component as XComponent)[XComponentModule] ?? null;
}
```

We can't pass a possible `null` value to the getXComponentXModuleName. I can modify it but we'd to the same as this line.